public/viewjs/consume.js

```diff
@@ -476,13 +476,13 @@ Grocy.Components.ProductPicker.GetPicker().on('change', function(e)
 
 				if (productDetails.product.enable_tare_weight_handling == 1)
 				{
-					$("#display_amount").attr("min", productDetails.product.tare_weight);
-					$('#display_amount').attr('max', productDetails.stock_amount + productDetails.product.tare_weight);
+					$("#display_amount").attr("min", productDetails.product.tare_weight * $("#qu_id option:selected").attr("data-qu-factor"));
+					$("#display_amount").attr("max", (productDetails.stock_amount + productDetails.product.tare_weight) * $("#qu_id option:selected").attr("data-qu-factor"));
```

Comment:
In this case, we need to multiply the tare weight by the conversion factor as well, since the amount given by the user contains it, but that may be specified in a quantity unit other than the stock QU.

Reply:
As of today, by definition the product's tare weight is related to the product's QU stock (and that's exactly outlined like so on the product edit page). So if you want to make the tare weight "selected QU aware", feel free to do so, but then by far this here is not the only place to take care of.